test/shared/topology_test.exs

```diff
@@ -130,6 +139,34 @@ defmodule Firenest.TopologyTest do
     end
 
     @node :"sync_named@127.0.0.1"
+    test "links to synced process", config do
+      %{topology: topology, evaluator: evaluator, test: test} = config
+      parent = self()
+
+      Firenest.Test.spawn_nodes([@node])
+      Firenest.Test.start_firenest([@node], adapter: T.adapter!(topology))
+
+      assert {:ok, []} = T.sync_named(topology, self())
+      start_sync_named_on(topology, @node, evaluator, test)
+      assert_receive {:named_up, @node, node_id, ^test}
+
+      cmd =
+        quote do
+          # Application.ensure_all_started(:sasl)
+          ref = Process.monitor(unquote(test))
+          Process.exit(Process.whereis(unquote(topology)), :kill)
+
+          receive do
+            {:DOWN, ^ref, _, _, _} ->
+              send(unquote(parent), :success)
```

Comment:
This uses a regular send to relay information that everything succeeded. I'm not really sure what to do otherwise since the topology in this node is down, so we can't use that to relay that information.

Unfortunately this means this test assumes erlang distribution under the topology.

Reply:
Those are shared topology tests, so we can't really do this. There are other things you could try to do, like writing a file to disk.

Reply:
on second thought, it uses the `:slave` instrumentation to start the node, so we always know we have a disterl connection to it besides the topology one, right?

Reply:
@michalmuskala I think using disterl to start the nodes is optional, you don't need to use `test.ex`, that's why it is part of the test helper and not of part of test per se.

I would say it is even preferrerable to not depend on disterl when testing other topologies because things may work by accident (i.e. we can accidentally use `send` in some of our code and it would magically work).

Reply:
@michalmuskala you are correct. Go ahead then. :)